site/gulpfile.js

```diff
@@ -638,7 +638,14 @@ const generateView = () => {
 
     // Get the list of codelabs and categories for this view
     const filtered = filterCodelabs(view, all.codelabs);
-    const codelabs = filtered.codelabs;
+    let codelabs = filtered.codelabs;
+    // Convert all codelab IDs from underscore(_) to dash(-)
+    codelabs = codelabs.map(codelab => {
+      codelab.id = codelab.id.split('_').join('-')
+      codelab.url = codelab.url.split('_').join('-')
```

Comment:
@loopDelicious this line also ensures that the URL always uses hyphen even if the id uses an underscore.